src/dronevis/drone_connect/nav_data_decode.py

```diff
@@ -35,75 +37,83 @@ def _drone_status_decode(packet):
+    navdata_demo["psi"] = struct.unpack_from("=f", packet[2], 16)[0]
+    navdata_demo["altitude"] = struct.unpack_from("=i", packet[2], 20)[0]
+    navdata_demo["vx"] = int(struct.unpack_from("=f", packet[2], 24)[0])
+    navdata_demo["vy"] = int(struct.unpack_from("=f", packet[2], 28)[0])
+    navdata_demo["vz"] = int(struct.unpack_from("=f", packet[2], 32)[0])
     return navdata_demo
 
+
 def _vision_detect_decode(packet):
     "Decode the vision detection packet, packet is (id=16, size, data)"
     if packet[0] != 16:
         raise IOError("Packet is not vision-detect packet")
     # Have to check if there isn't tag and size first
-    vision_detect = dict()
-    vision_detect["nb_detected"] = struct.unpack_from("=I",packet[2][0:20])[0]
-    vision_detect["xc"] = struct.unpack_from("=I",packet[2][20:36])[0]
-    vision_detect["yc"] = struct.unpack_from("=I",packet[2][36:52])[0]
-    vision_detect["width"] = struct.unpack_from("=I",packet[2][52:68])[0]
-    vision_detect["height"] = struct.unpack_from("=I",packet[2][68:84])[0]
-    vision_detect["distance"] = struct.unpack_from("=I",packet[2][84:100])[0]
+    vision_detect = {}
+    vision_detect["nb_detected"] = struct.unpack_from("=I", packet[2][0:20])[0]
+    vision_detect["xc"] = struct.unpack_from("=I", packet[2][20:36])[0]
+    vision_detect["yc"] = struct.unpack_from("=I", packet[2][36:52])[0]
+    vision_detect["width"] = struct.unpack_from("=I", packet[2][52:68])[0]
+    vision_detect["height"] = struct.unpack_from("=I", packet[2][68:84])[0]
+    vision_detect["distance"] = struct.unpack_from("=I", packet[2][84:100])[0]
     return vision_detect
 
+
 def _gps_decode(packet):
     "Decode data about the GPS"
-    gps_info = dict()
-    if packet[0]!=27:
+    gps_info = {}
+    if packet[0] != 27:
         raise IOError("Packet isn't navdata-demo packet")
-    unpacked = struct.unpack_from("=ddddBBB",packet[2],0)
-    gps_info["latitude"]= unpacked[0]
-    gps_info["longitude"]= unpacked[1]
-    gps_info["elevation"]= unpacked[2]
-    gps_info["hdop"]= unpacked[3]
-    gps_info["data_available"]= unpacked[4]
-    gps_info["zero_validated"]= unpacked[5]
-    gps_info["wpt_validated"]= unpacked[6]
+    unpacked = struct.unpack_from("=ddddBBB", packet[2], 0)
+    gps_info["latitude"] = unpacked[0]
+    gps_info["longitude"] = unpacked[1]
+    gps_info["elevation"] = unpacked[2]
+    gps_info["hdop"] = unpacked[3]
+    gps_info["data_available"] = unpacked[4]
+    gps_info["zero_validated"] = unpacked[5]
+    gps_info["wpt_validated"] = unpacked[6]
     return gps_info
-    
+
+
 def navdata_decode(packet):
     "Split then decodes the navdata packet gathered from UDP 5554"
-    position=0
+    position = 0
     offset = 0
-    block=[]
-    block.append(struct.unpack_from("=IIII",packet,position))
+    block = []
+    block.append(struct.unpack_from("=IIII", packet, position))
     offset += struct.calcsize("=IIII")
-    i=1
+    i = 1
 
     while 1:
```

Comment:
I think it should be while True to be consistent with other files in the project.